src/sentry/static/sentry/app/components/events/interfaces/spans/spanTree.tsx

```diff
@@ -0,0 +1,547 @@
+          </span>
+          <div style={{marginRight: '2px', width: '5px', textAlign: 'right'}}>
+            {chevron}
+          </div>
+        </SpanTreeToggler>
+      </SpanTreeTogglerContainer>
+    );
+  };
+
+  renderTitle = () => {
+    const {span, treeDepth} = this.props;
+
+    const op = span.op ? <strong>{`${span.op} \u2014 `}</strong> : '';
+    const description = get(span, 'description', span.span_id);
+
+    const MARGIN_LEFT = 8;
+    const TOGGLE_BUTTON_MARGIN_RIGHT = 8;
+    const TOGGLE_BUTTON_MAX_WIDTH = 40;
+
+    const left =
+      treeDepth * (TOGGLE_BUTTON_MAX_WIDTH + TOGGLE_BUTTON_MARGIN_RIGHT) + MARGIN_LEFT;
+
+    return (
+      <SpanBarTitleContainer>
+        {this.renderSpanTreeToggler({left})}
+        <SpanBarTitle
+          data-component="span-bar-title"
+          style={{
+            left: `${left}px`,
+            width: '100%',
+          }}
+        >
+          <span>
+            {op}
+            {description}
+          </span>
+        </SpanBarTitle>
+      </SpanBarTitleContainer>
+    );
+  };
+
+  renderSpanChildren = () => {
+    if (!this.state.showSpanTree) {
+      return null;
+    }
+
+    return this.props.renderedSpanChildren;
+  };
+
+  render() {
+    const {span, spanBarColour} = this.props;
+
+    const startTimestamp: number = span.start_timestamp;
+    const endTimestamp: number = span.timestamp;
+
+    const duration = Math.abs(endTimestamp - startTimestamp);
+    const durationString = getHumanDuration(duration);
+
+    const bounds = this.getBounds();
+
+    const isVisible = bounds.end > 0 && bounds.start < 1;
+
+    return (
+      <React.Fragment>
+        <SpanRow
+          data-span-hidden={isVisible ? 'false' : 'true'}
+          style={{
+            display: isVisible ? 'block' : 'none',
+            boxShadow: this.state.displayDetail ? '0 -1px 0 #d1cad8' : void 0,
+          }}
+          onClick={() => {
+            this.toggleDisplayDetail();
+          }}
+        >
+          <SpanBar
+            data-span="true"
+            style={{
+              backgroundColor: spanBarColour,
+              left: toPercent(bounds.start),
+              width: toPercent(bounds.end - bounds.start),
+            }}
+          />
+          {this.renderTitle()}
+          <Duration>{durationString}</Duration>
+          {this.renderDetail({isVisible})}
+        </SpanRow>
+        {this.renderSpanChildren()}
+      </React.Fragment>
+    );
+  }
+}
+
+const TraceViewContainer = styled('div')`
+  overflow-x: hidden;
+  border-bottom-left-radius: 3px;
+  border-bottom-right-radius: 3px;
+`;
+
+const SPAN_ROW_HEIGHT = 25;
+
+const SpanRow = styled('div')`
+  position: relative;
+  overflow: hidden;
+
+  cursor: pointer;
+  transition: background-color 0.15s ease-in-out;
+
+  &:last-child {
+    & > [data-component='span-detail'] {
+      border-bottom: none !important;
+    }
+  }
+
+  &:hover {
+    background-color: rgba(189, 180, 199, 0.1);
+
+    & > [data-span='true'] {
```

Comment:
e.g. for this we could have in `SpanBar` 

```
${SpanRow}:hover & {
    transition: border-color 0.15s ease-in-out;
    border: 1px solid rgba(0, 0, 0, 0.1);
}
```

Otherwise we have another set of attributes to keep track of (and there's no build time checks for data attribute refereces)